frontend/src/components/ChangeLog/ChangeLogTable.tsx

```diff
@@ -0,0 +1,299 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Box,
+  IconButton,
+  CircularProgress,
+  alpha,
+  Typography,
+  TablePagination,
+  TextField,
+  Button,
+} from '@mui/material';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import ExpandLessIcon from '@mui/icons-material/ExpandLess';
+import type { ChangeLogEntry } from 'models/ChangeLogEntry';
+import { useTranslation } from 'react-i18next';
+import { formatTimestamp, getVacancyNumber } from './utils';
+import { useGetPositionsQuery } from 'redux/api-slices/functions/positions-api';
+import { useGetChangeLogsQuery } from 'redux/api-slices/functions/changelogs-api';
+
+const ChangeLogTable = () => {
+  const { t } = useTranslation();
+
+  const [expandedRow, setExpandedRow] = useState<string | null>(null);
+  const [filteredChangeLogs, setFilteredChangeLogs] = useState<ChangeLogEntry[]>([]);
+  const [sortConfig, setSortConfig] = useState<{ key: keyof ChangeLogEntry; direction: 'asc' | 'desc' } | null>(null);
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const [searchVacancyNumber, setSearchVacancyNumber] = useState('');
+  const [searchDate, setSearchDate] = useState('');
+
+  const { data: positions = [], isLoading: positionsLoading } = useGetPositionsQuery();
+  const { data: changeLogs = [], isLoading: changeLogsLoading } = useGetChangeLogsQuery();
+
+  const isLoading = positionsLoading || changeLogsLoading;
+
+  useEffect(() => {
+    setFilteredChangeLogs(changeLogs);
+  }, [changeLogs]);
+
+  const getTranslatedField = (field: string) => {
+    return t(`change_logs.fields.${field}`, field);
+  };
+
+  // Sorting logic
+  const handleSort = (key: keyof ChangeLogEntry) => {
+    setSortConfig((prevConfig) => {
+      if (prevConfig && prevConfig.key === key) {
+        return { key, direction: prevConfig.direction === 'asc' ? 'desc' : 'asc' };
+      }
+      return { key, direction: 'asc' };
+    });
+  };
+
+  const sortedChangeLogs = React.useMemo(() => {
+    if (!sortConfig) return filteredChangeLogs;
+    return [...filteredChangeLogs].sort((a, b) => {
+      const aValue = a[sortConfig.key];
+      const bValue = b[sortConfig.key];
+
+      if (typeof aValue === 'string' && typeof bValue === 'string') {
+        return sortConfig.direction === 'asc' ? aValue.localeCompare(bValue) : bValue.localeCompare(aValue);
+      }
+      if (typeof aValue === 'number' && typeof bValue === 'number') {
```

Comment:
`aValue` and `bValue` are always typeof string, so this is invalid. (Means also that the above if statement checks type for no reason.

Reply:
True, I'll fix that